crates/core_simd/src/vector.rs

```diff
@@ -492,6 +622,31 @@ where
         unsafe { core::intrinsics::simd::simd_gather(or, source, enable.to_int()) }
     }
 
+    #[inline]
+    pub fn masked_store(self, slice: &mut [T], mut enable: Mask<<T as SimdElement>::Mask, N>) {
+        enable &= mask_up_to(enable, slice.len());
```

Comment:
Would `mask_up_to()` already do "and" ops to the input `enable`? 

```rust
enable & Mask::<M, N>::from_bitmask_vector(lt.to_bitmask_vector())
```
i.e. it could be `enable = mask_up_to(enable, slice.len());` here. 

(or dropping `enable &` from function `mask_up_to()`)